test/test_utils/benchmark.py

```diff
@@ -0,0 +1,91 @@
+import os
+
+
+def single_node_benchmark_executor(context, image_uri, framework, task, py_version):
+    """
+    Shared code to execute single-node benchmarks
+    :param context: Context() from callers
+    :param image_uri: identifies mage
+    :param framework: Framework ex(TF,MXNET)
+    :param task: specifies task name
+    :param py_version: python version
+    """
+    benchmark_client_id = os.getenv("BENCHMARK_CLIENT_ID")
+    benchmark_endpoint = os.getenv("BENCHMARK_ENDPOINT")
+    benchmark_exec = os.getenv("BENCHMARK_EXEC")
+    container_name = f"{framework}_single_node-{image_uri.split('/')[-1].replace('.', '-').replace(':', '-')}"
+    bai_dir = os.path.join(os.getcwd(), "benchmark", "bai")
+    path_to_toml = os.path.join(bai_dir, framework, "training", f"{framework}_single.toml")
+    custom_toml_name = f"custom_{framework}_{py_version}.toml"
+    write_image_to_toml(image_uri, path_to_toml, task, custom_toml_name)
+    toml = os.path.join("test", framework, "training", custom_toml_name)
+    context.run(
+        f"docker run --name {container_name} -v {bai_dir}:{os.path.join(os.sep, 'test')} -itd bai_env_container",
+        hide=True,
+    )
+    if framework == "mxnet":
+        execute_bash_command(
+            context, container_name, f"git clone https://github.com/awslabs/deeplearning" f"-benchmark.git"
+        )
+    elif framework == "tensorflow":
+        execute_bash_command(context, container_name, f"git clone https://github.com/tensorflow" f"/benchmarks.git")
+
+    execute_bash_command(
+        context, container_name, f"curl http://{benchmark_endpoint}/api/tools/{benchmark_exec} " f"-o {benchmark_exec}"
+    )
+    execute_bash_command(context, container_name, f"chmod u+x {benchmark_exec}")
+    execute_bash_command(context, container_name, f"./{benchmark_exec} --env-setup")
+    execute_bash_command(context, container_name, f"./{benchmark_exec} --register {benchmark_endpoint}")
+    execute_bash_command(context, container_name, f"./{benchmark_exec} --set-client-id {benchmark_client_id}")
+    output = context.run(
+        f"docker exec --user root {container_name} bash -c './{benchmark_exec} --get-client-id'", hide=True,
+    )
+    assert benchmark_client_id in output.stdout, output.stdout
+    if framework == "mxnet":
```

Comment:
is the only difference between the mxnet and tf conditions the script? if so, maybe it makes sense to add a parameter for script to avoid having to make unique conditions with hardcoded frameworks